### ISSUE TITLE
Clarification on BulkheadSynchTest in FT 2.0

### ISSUE THREAD

**Issue by spericas** [eclipse/microprofile-fault-tolerance#420]
One of the changes in the FT 2.0 TCK is a new implementation of `bulkhead.Utils::handleResults(...)` which _now fails if an exception is thrown_ while getting the value out of a `Future`.

Test method `BulkheadSyncTest::testBulkheadClassSemaphore3()` creates 20 threads to access a bulkhead of size 3. The futures created for this test are passed to `Utils::handleResults(...)` for processing. 

 Why is a `BulkheadException` _not expected_ while processing these futures in `Utils::handleResults(...)` given the large number of threads? Perhaps I missed some change/clarification to the spec.  Thanks.

**Comment by Azquelt** [comment]
In that particular test, it you check through the `threads()` method, you should be able to see that each `Future` is obtained by submitting a `org.eclipse.microprofile.fault.tolerance.tck.bulkhead.clientserver.ParrallelBulkheadTest` to run asyncrhonously. `ParrallelBulkheadTest.call()` has a catch block to handle a `BulkheadException` so that exception shouldn't ever end up in the `Future`.

I'm not a huge fan of those tests as I find them hard to follow and find it difficult to see what's actually being asserted, but I think that in this case it should be ok.

If you have a test where a `BulkheadException` is being found and reported inside `Utils::handleResults()`, I'd be happy to help debug it.

**Comment by Emily-Jiang** [comment]
@spericas are you happy with the explanation from @Azquelt? We are going to improve the test with more readability issue address under #271 . As of today's hangout, we plan to close this one and make some progress on #271. Please reopen if this does not address your concern.

**Comment by spericas** [comment]
@Emily-Jiang @Azquelt Thanks for the explanation. I agree the test is correct.
